Expensify/expensify-common
# fix: add support for localhost urls in markdown

This PR adds support for auto linking of local urls like `http://my-localhost`, `http://localhost:3000`

### Fixed Issues
$ https://github.com/Expensify/App/issues/17828

# Tests
1. What unit/integration tests cover your change? What autoQA tests cover your change?
`URL-test`, `ExpensiMark-HTML-test` and `ExpensiMark-HTMLToText-test`

1. What tests did you perform that validates your changed worked?
Added new test cases that auto link local urls.

# QA
1. What does QA need to do to validate your changes?
Run the new test cases

1. What areas to they need to test for regressions?
Check if urls are getting auto linked correctly
